com.unity.render-pipelines.universal/Runtime/ScriptableRenderer.cs

```diff
@@ -78,11 +78,15 @@ public static void SetCameraMatrices(CommandBuffer cmd, ref CameraData cameraDat
 
             if (setInverseMatrices)
             {
+                Matrix4x4 gpuProjectionMatrix = cameraData.GetGPUProjectionMatrix();
+                Matrix4x4 viewAndProjectionMatrix = gpuProjectionMatrix * viewMatrix;
                 Matrix4x4 inverseViewMatrix = Matrix4x4.Inverse(viewMatrix);
-                cmd.SetGlobalMatrix(ShaderPropertyId.cameraToWorldMatrix, inverseViewMatrix);
-
-                Matrix4x4 viewAndProjectionMatrix = cameraData.GetGPUProjectionMatrix() * viewMatrix;
+                Matrix4x4 inverseProjectionMatrix = Matrix4x4.Inverse(gpuProjectionMatrix);
                 Matrix4x4 inverseViewProjection = Matrix4x4.Inverse(viewAndProjectionMatrix);
+
+                cmd.SetGlobalMatrix(ShaderPropertyId.cameraToWorldMatrix, inverseViewMatrix);
+                cmd.SetGlobalMatrix(ShaderPropertyId.inverseViewMatrix, inverseViewMatrix);
```

Comment:
If I understand this correctly we have two properties mapping to same inverseViewMatrix value?
 If so we could have a define in shader library to forward/map one to another instead of making making redundant set uniform calls.

Reply:
I think `ShaderPropertyId.cameraToWorldMatrix` refers to the camera matrices, whereas `ShaderPropertyId.inverseViewMatrix` refers to the pass matrices.

This function is setting both camera matrices and per-pass matrices.

Reply:
ah right.